sys/vm_map.c

```diff
@@ -212,36 +239,90 @@ void vm_map_dump(vm_map_t *map) {
   vm_map_entry_t *it;
   kprintf("[vm_map] Virtual memory map (%08lx - %08lx):\n", map->pmap->start,
           map->pmap->end);
+  rw_enter(&map->rwlock, RW_READER);
   TAILQ_FOREACH (it, &map->list, map_list) {
     kprintf("[vm_map] * %08lx - %08lx [%c%c%c]\n", it->start, it->end,
             (it->prot & VM_PROT_READ) ? 'r' : '-',
             (it->prot & VM_PROT_WRITE) ? 'w' : '-',
             (it->prot & VM_PROT_EXEC) ? 'x' : '-');
     vm_map_object_dump(it->object);
   }
+  rw_leave(&map->rwlock);
+}
+
+vm_addr_t vm_map_sbrk(vm_map_t *map, intptr_t increment) {
```

Comment:
This procedure does too much. Please move `sbrk` segment creation to `exec` syscall code. Then I'll revisit `sbrk` implementation.